ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala

```diff
@@ -624,45 +569,61 @@ object LedgerApiErrors extends LedgerApiErrorGroup {
-      ) {
-    case class Error(
-        fieldName: String,
-        offsetValue: String,
-        message: String,
-    )(implicit
-        override val loggingContext: ContextualizedErrorLogger
-    ) extends BaseError.Impl(
-          cause = s"Offset in ${fieldName} not specified in hexadecimal: ${offsetValue}: ${message}"
-        )
+    case class VersionService(message: String)(implicit
+        loggingContext: ContextualizedErrorLogger
+    ) extends LoggingTransactionErrorImpl(cause = message)
   }
 
-  object VersionServiceError extends ErrorGroup {
-    @Explanation("This error occurs if there was an unexpected error within the version service.")
-    @Resolution("Contact support.")
-    object InternalError
+  object AdminServices {
+    @Explanation("This rejection is given when a configuration entry write was rejected.")
+    @Resolution("Fetch newest configuration and/or retry.")
+    object ConfigurationEntryRejected
         extends ErrorCode(
-          id = "VERSION_SERVICE_INTERNAL_ERROR",
-          ErrorCategory.SystemInternalAssumptionViolated,
+          id = "CONFIGURATION_ENTRY_REJECTED",
+          ErrorCategory.InvalidGivenCurrentSystemStateOther,
         ) {
+      case class Reject(message: String)(implicit
+          loggingContext: ContextualizedErrorLogger
+      ) extends LoggingTransactionErrorImpl(
+            cause = message
+          )
+    }
 
+    @Explanation("This rejection is given when a package upload was rejected.")
+    @Resolution("Refer to the detailed message of the received error.")
+    object PackageUploadRejected
+        extends ErrorCode(
+          id = "PACKAGE_UPLOAD_REJECTED",
+          ErrorCategory.InvalidGivenCurrentSystemStateOther,
+        ) {
       case class Reject(message: String)(implicit
           loggingContext: ContextualizedErrorLogger
-      ) extends LoggingTransactionErrorImpl(cause = message)
+      ) extends LoggingTransactionErrorImpl(
+            cause = message
+          )
+    }
+
+    @Explanation(
+      "This rejection is given when a request might not been processed and a time-out was reached."
+    )
+    @Resolution(
+      "Retry for transient problems. If non-transient contact the operator as the time-out limit might be to short."
+    )
+    object RequestTimeOut
+        extends ErrorCode(
+          id = "REQUEST_TIME_OUT",
+          ErrorCategory.DeadlineExceededRequestStateUnknown,
+        ) {
+      case class Reject(message: String, override val definiteAnswer: Boolean)(implicit
+          loggingContext: ContextualizedErrorLogger
+      ) extends LoggingTransactionErrorImpl(
+            cause = message
+          )
     }
   }
 
-  object CommandRejections extends ErrorGroup {
+  object ConsistencyErrors extends ErrorGroup {
     @Explanation("A command with the given command id has already been successfully processed")
     @Resolution("Celebrate, as your command has already been delivered")
```

Comment:
Nit-picking:
Not sure if the humor is appropriate. I guess there might be a scenario when by reading this resolution message the user might get more annoyed/irritated.

Reply:
Yeah, noticed this one. Will change.

Reply:
Changed to 
```
    @Resolution(
      """The correct resolution depends on the use case. If the error received pertains to a submission retried due to a timeout,
        |do nothing, as the a previous command has already been accepted.
        |If the intent is to submit a new command, re-submit using a distinct command id. 
        |""".stripMargin
```